reducer/src/main/java/com/graphicsfuzz/reducer/reductionopportunities/UnwrapReductionOpportunities.java

```diff
@@ -99,21 +104,31 @@ public void visitDoStmt(DoStmt doStmt) {
 
   @Override
   protected void visitChildOfBlock(BlockStmt block, Stmt child) {
-    if (isNonEmptyBlockStmtWithoutTopLevelDeclaration(child)) {
-      addOpportunity(
-            new UnwrapReductionOpportunity(child, ((BlockStmt) child).getStmts(),
-                  parentMap,
-                  getVistitationDepth()));
+    if (!(child instanceof BlockStmt)) {
+      return;
     }
-  }
-
-  private boolean isNonEmptyBlockStmtWithoutTopLevelDeclaration(Stmt stmt) {
-    if (!(stmt instanceof BlockStmt)) {
-      return false;
+    final BlockStmt childAsBlock = (BlockStmt) child;
+    if (childAsBlock.getNumStmts() == 0) {
+      // We do not unwrap an empty block statement, as there would be nothing to unwrap.
+      // Other reductions take responsibility for removing a redundant { }.
+      return;
+    }
+    // If unwrapping the child block may lead to name collisions then it cannot be done.
```

Comment:
I suppose this could cause issues, but this is not a new issue:

```c
void main() {
  {
      int max;
  }

  max(1.0, 1.0);
}

```

Reply:
Agreed; I have filed #408 for this.